graveler/graveler.go

```diff
@@ -1327,3 +1352,205 @@ func (g *Graveler) getCommitsForMerge(ctx context.Context, repositoryID Reposito
 	}
 	return fromCommit, toCommit, baseCommit, nil
 }
+
+func (g *Graveler) DumpCommits(ctx context.Context, repositoryID RepositoryID) (*MetaRangeID, error) {
+	repo, err := g.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return nil, err
+	}
+	iter, err := g.RefManager.ListCommits(ctx, repositoryID)
+	if err != nil {
+		return nil, err
+	}
+	defer iter.Close()
+	mid, err := g.CommittedManager.WriteMetaRange(ctx, repo.StorageNamespace,
+		commitsToValueIterator(iter),
+		Metadata{"entity": "commit"},
```

Comment:
1. Done
2. Not at the moment. Since metadata is captured as part of the content address, I'm a little reluctant to add it there.
3. Added both fully qualified name of proto message and a JSON encoded proto message of the descriptor. (JSON, because sstable UserProperties accepts strings and I'm not crazy about sticking non UTF-8 bytes into it.